grammars/javascript.cson

```diff
@@ -702,14 +702,21 @@
   {
     # console
     # console.log(arg1, "arg2", [...])
-    'begin': '\\bconsole\\b'
+    'begin': '(?<![\\w$])console(?![\\w$]|\\s*:)'
     'beginCaptures':
       '0':
         'name': 'entity.name.type.object.console.js'
-    'end': '(?!\\G)'
+    'end': '''(?x)
+      (?=
+        (?! \\s*(\\.)\\s*
+          (assert|clear|debug|error|info|log|profile|profileEnd|time|timeEnd|warn)
+          \\s*\\(
+        ) \\S
+      )
+    '''
     'patterns': [
       {
-        'begin': '\\s*(\\.)\\s*(assert|clear|debug|error|info|log|profile|profileEnd|time|timeEnd|warn)\\s*(?=\\()'
+        'begin': '\\s*(\\.)\\s*(\\w+)\\s*(?=\\()'
```

Comment:
`\w` is enough here, as I modified `end` pattern to allow regular functions/properties.